From 0775c18cbe96ee6ed779fbaf98c6fca65cdc6d12 Mon Sep 17 00:00:00 2001
From: Sultan Alsawaf <sultan@kerneltoast.com>
Date: Sat, 16 Dec 2023 11:05:33 -0800
Subject: [PATCH 089/105] modesetting: Enable TearFree by default

TearFree support has been available in the modesetting driver for a year
with no issues reported. The code is mature and robust, with error handling
that's been vetted across many hardware configurations.

Notably, TearFree is also the only way to achieve a tear-free desktop with
mismatched displays and transformed CRTCs.

Enable TearFree by default for a smooth desktop experience out of the box.

Signed-off-by: Sultan Alsawaf <sultan@kerneltoast.com>
---
 hw/xfree86/drivers/modesetting/driver.c        | 2 +-
 hw/xfree86/drivers/modesetting/modesetting.man | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw/xfree86/drivers/modesetting/driver.c b/hw/xfree86/drivers/modesetting/driver.c
index 334102d9c..c8542b89f 100644
--- a/hw/xfree86/drivers/modesetting/driver.c
+++ b/hw/xfree86/drivers/modesetting/driver.c
@@ -1381,7 +1381,7 @@ PreInit(ScrnInfoPtr pScrn, int flags)
                "Atomic modesetting %sabled\n", ms->atomic_modeset ? "en" : "dis");
 
     /* TearFree requires glamor and, if PageFlip is enabled, universal planes */
-    if (xf86ReturnOptValBool(ms->drmmode.Options, OPTION_TEARFREE, FALSE)) {
+    if (xf86ReturnOptValBool(ms->drmmode.Options, OPTION_TEARFREE, TRUE)) {
         if (pScrn->is_gpu) {
             xf86DrvMsg(pScrn->scrnIndex, X_WARNING,
                        "TearFree cannot synchronize PRIME; use 'PRIME Synchronization' instead\n");
diff --git a/hw/xfree86/drivers/modesetting/modesetting.man b/hw/xfree86/drivers/modesetting/modesetting.man
index 485beff48..4a8af97c8 100644
--- a/hw/xfree86/drivers/modesetting/modesetting.man
+++ b/hw/xfree86/drivers/modesetting/modesetting.man
@@ -118,7 +118,7 @@ as rotated and scaled CRTCs. When PageFlip is enabled, fullscreen DRI
 applications will still have the discretion to not use tearing prevention.
 .br
 The default is
-.B off.
+.B on.
 .TP
 .BI "Option \*qAtomic\*q \*q" boolean \*q
 Enable atomic modesetting when supported.  The default is
-- 
2.45.2

